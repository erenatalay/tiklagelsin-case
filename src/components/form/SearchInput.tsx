import React, { FC } from 'react'
import { StyleSheet, useColorScheme } from 'react-native';
import { Searchbar, SearchbarProps, DefaultTheme, TextInput, TextInputProps } from 'react-native-paper'
import Colors from '../../constant/Colors';

interface SearchInputProps extends TextInputProps {

}

const SearchInput: FC<SearchInputProps> = (props) => {
    const color = useColorScheme() ?? 'light';

    const theme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: "white",
            text: "blue",
        },
        ...{ roundness: 30 },

    };
    return (
        <TextInput
            theme={theme}
            left={<TextInput.Icon icon={"magnify"} />}
            style={styles.textInput}
            placeholderTextColor={Colors[color].textInputPlaceholder}
            activeOutlineColor={Colors[color].textInputBackground}
            mode={"outlined"}
            {...props}
        />
    )
}

const styles = StyleSheet.create({
 textInput : {
        paddingVertical: 12,
        borderRadius: 64,
        height: 20,
        justifyContent: "center",
        marginHorizontal : 15
 }
})

export default SearchInput