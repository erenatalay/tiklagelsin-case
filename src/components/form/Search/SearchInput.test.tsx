import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SearchInput from './SearchInput';
describe('Input component tests', () => {
    it('is rendered correctly', () => {
        expect(<SearchInput
            placeholder="Enter text"
        />
        ).toMatchSnapshot();
    });

    it('should trigger Password when typed', () => {
        const mockSearch = jest.fn();
        const wrapper = render(
            <SearchInput
                placeholder="Enter text"
                onChangeText={mockSearch}
            />
        );
        const input = wrapper.getByTestId('searchbar-input');
        fireEvent(input, 'onChangeText', 'test');
        expect(mockSearch).toHaveBeenCalled();
    });
});

