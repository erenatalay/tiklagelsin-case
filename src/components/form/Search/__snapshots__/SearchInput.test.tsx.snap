// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests is rendered correctly 1`] = `
<SearchInput
  placeholder="Enter text"
/>
`;
